frontline-functions-quickstart/functions/callbacks/templates.protected.js

```diff
@@ -0,0 +1,101 @@
+  'Just to confirm I am on my way to your office. {{Name}}.';
+
+const REPLY_SENT = 'This has now been sent. {{Author}}.';
+const REPLY_RATES =
+  'Our rates for any loan are 20% or 30% over $30,000. You can read more at https://example.com. {{Author}}.';
+const REPLY_OMW = 'Just to confirm I am on my way to your office. {{Author}}.';
+const REPLY_OPTIONS =
+  'Would you like me to go over some options with you {{Name}}? {{Author}}.';
+const REPLY_ASK_DOCUMENTS =
+  'We have a secure drop box for documents. Can you attach and upload them here: https://example.com. {{Author}}';
+
+const CLOSING_ASK_REVIEW =
+  'Happy to help, {{Name}}. If you have a moment could you leave a review about our interaction at this link: https://example.com. {{Author}}.';
+
+const handleGetTemplatesByCustomerIdCallback = async (context, event) => {
+  console.log('Getting templates: ', event.CustomerId);
+
+  const customerDetails = await getCustomerById(context, event.CustomerId);
+
+  if (!customerDetails) {
+    throw new Error('Customer not found');
+  }
+
+  // Prepare templates categories
+  const openersCategory = {
+    display_name: 'Openers', // Category name
+    templates: [
+      { content: compileTemplate(OPENER_NEXT_STEPS, customerDetails) }, // Compiled template
+      { content: compileTemplate(OPENER_NEW_PRODUCT, customerDetails) },
+      {
+        content: compileTemplate(OPENER_ON_MY_WAY, customerDetails),
+        whatsAppApproved: true,
+      },
+    ],
+  };
+  const repliesCategory = {
+    display_name: 'Replies',
+    templates: [
+      { content: compileTemplate(REPLY_SENT, customerDetails) },
+      { content: compileTemplate(REPLY_RATES, customerDetails) },
+      { content: compileTemplate(REPLY_OMW, customerDetails) },
+      { content: compileTemplate(REPLY_OPTIONS, customerDetails) },
+      { content: compileTemplate(REPLY_ASK_DOCUMENTS, customerDetails) },
+    ],
+  };
+  const closingCategory = {
+    display_name: 'Closing',
+    templates: [
+      { content: compileTemplate(CLOSING_ASK_REVIEW, customerDetails) },
+    ],
+  };
+
+  // Respond with compiled Templates
+  return [openersCategory, repliesCategory, closingCategory];
+};
+
+exports.handler = async function (context, event, callback) {
+  console.log('[[ Templates handler ]]');
+  try {
+    console.log(event);
+    const location = event.Location;
+
+    /**
+     * Location helps to determine which information was requested.
+     * callback is a general purpose tool and might be used to fetch different kind of information
+     */
+    if (location === 'GetTemplatesByCustomerId') {
+      console.log('[[[ GetTemplatesByCustomerId ]]]');
+      try {
+        const resp = await handleGetTemplatesByCustomerIdCallback(
+          context,
+          event
+        );
+        return callback(null, resp);
+      } catch (err) {
+        console.log(err);
+        return callback(err);
+      }
+    }
+
+    return callback(422, `Unknown location: location`);
```

Comment:
I guess it supposed to be `Unknown location: ${location}`?

Reply:
😄 yes!